src/common/njord.ts

```diff
@@ -709,12 +763,34 @@ export const awardComment = async (
         await saveComment(entityManager, newComment, post.sourceId);
       }
 
-      const transfer = await transferCores({
-        ctx,
-        transaction,
-      });
+      try {
+        const transfer = await transferCores({
+          ctx,
+          transaction,
+        });
 
-      return { transaction, transfer };
+        return { transaction, transfer };
+      } catch (error) {
+        if (error instanceof TransferError) {
+          if (newComment) {
+            await entityManager.getRepository(Comment).delete({
+              id: newComment.id,
+            });
+          }
+
+          await entityManager.getRepository(UserComment).delete({
+            awardTransactionId: transaction.id,
+          });
```

Comment:
Should we use transaction for both?

Reply:
it is in transaction

Reply:
yes i missed it 🤦 
i always assume the `entityManager` syntax is out of transaction for some reaosn